### DISCUSSION TITLE
"Uncaught Snapshot missing on Livewire component with id: *" with child component and wire:key

### DISCUSSION THREAD

**Discussion by timetorock** [livewire/livewire#8575] [7. Bugs (archived)] [answered]
### Livewire version

v3.5.0

### Laravel version

v10.35.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create parent component with elements + child that edit those elements. In my example TODOs component.
2. Set wire:key for each element.
3. Edit todo in child component, so update request sent and `wire:id` only left.
4. Try to add new todo in reverse order (array_unshift).
5. Uncaught Snapshot missing on Livewire component with id: 9p1S6fdShnbfDigB721P
6. Livewire stop working after this error.

P.S. Without editing child component, e.g. any requests to backend, adding new items working just fine. Problem is only for cases where we need to update child first, and then render parent again with new items and order.

### Repository or Wirebox URL

https://wirebox.app/b/4lk6d

### Additional information

In reality my component is far more complex than this one, but I've been able to trigger same behaviour. Main idea is that in my component I show last created items first. That's why I need to order them like that in wirebox example.

I have a lot of different wire:key on elements, but this didn't help, as in this example.

Can anyone suggest how I can avoid such problems? I've examined livewire repository, but it's really complex to provide fast solution in this. So maybe someone more experienced can help with this?

Broken code can be found in [wirebox.app](https://wirebox.app/b/4lk6d), I can provide more code if need, as we have this problem for a long time from first 3.* versions. I though it can be fixed in 3.5, but no, same behaviour.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by RoccoHoward** [comment]
The only way I got around this issue (had it pop up today) was to  make the wire:key generate new keys when the parent was rendered.

I simply added time() to the key value, so that refreshes would generate entirely new keys.

From the last part of https://livewire.laravel.com/docs/nesting

**Reply by timetorock** [reply]
@RoccoHoward Hi! Thank you for your answer. If I change my code for every child component to  
`@livewire('my-todo-edit', ['todo' => $todo['todo']], key(sprintf('live-edit-%s-%s', $todo['id'], time())))`

I will get another similar error after the same steps:
`livewire.js?id=87e1046f:4485 Uncaught Component already initialized`

Did I understand you right? Here it's with the same example above, but updated with time():
https://wirebox.app/b/ydmog

**Comment by timetorock** [comment, marked as answer]
Ok, I just found out what was the problem, and it's funny that I've also accidentally reproduced it in Wirebox:

My parent component initiation in controller:
```
public function index(): View
{
    return view('user.payment-methods.index');
}
```

It's code:

```
@extends('adminlte::page')

@section('content')
    <div class="row">
        <div class="col-12">
            @livewire('user.payment-method.payment-methods')
        </div>
    </div>
@stop
```

All children and main component body, everything was inside of `@livewire('user.payment-method.payment-methods')`. So I totally forgot about adding key to this core `@livewire()` tag.

After adding `@livewire('user.payment-method.payment-methods', key('payment-method-list'))` everything instantly normalized.

Regarding Wirebox, it's using full page components, and though it suffers from the same problem as my case. I'm actually not sure how to fix it there.
